nixos/modules/services/web-apps/birdsitelive.nix

```
@@ -0,0 +1,447 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.birdsitelive;
+  settingsType = (pkgs.formats.json { }).type;
+
+  isSecret = v: with builtins; isAttrs v && v ? _secret && isString v._secret;
+  secret = with lib; mkOptionType {
+    name = "secret";
+    description = "secret value";
+    descriptionClass = "noun";
+    check = isSecret;
+    getSubOptions = prefix: prefix ++ [ "_secret" ];
+    nestedTypes = {
+      _secret = types.str;
+    };
+  };
+
+  toJsonSec =
+    let toJsonSec' = {}@args: v: with builtins;
+      let concatItems = concatStringsSep ",";
+      in if isSecret v then
+        ''"${hashString "sha256" v._secret}"''
+      else if isAttrs v then
+        "{${concatItems (lib.mapAttrsToList
+       (key: val: ''"${key}":${toJsonSec' args val}'') v)}}"
+      else if isList v then
+        "[${concatItems (map (toJsonSec' args) v)}]"
+      else
+        builtins.toJSON v;
+    in toJsonSec' { };
+
+  configFile = pkgs.runCommand "appsettings.json" {
+      value = toJsonSec cfg.settings;
+      passAsFile = [ "value" ];
+    } ''${pkgs.jq}/bin/jq . "$valuePath" >$out'';
```

```suggestion
  configFile = pkgs.runCommand "appsettings.json" {
      nativeBuildInputs = with pkgs; [ jq ];
      value = toJsonSec cfg.settings;
      passAsFile = [ "value" ];
    } "jq . "$valuePath" >$out";
```
this is better for cross.

---

Corrected. I’ll keep that in mind for the future.